### ISSUE TITLE
Tracking Issue for `transmute_generic_consts`

### ISSUE THREAD

**Issue by JulianKnodt** [rust-lang/rust#109929]
The feature gate for the issue is `#![feature(transmute_generic_consts)]`.

### About tracking issues

Tracking issues are used to record the overall progress of implementation.
They are also used as hubs connecting to other relevant issues, e.g., bugs or open design questions.
A tracking issue is however *not* meant for large scale discussion, questions, or bug reports about a feature.
Instead, open a dedicated issue for the specific matter and add the relevant feature gate label.

### Steps

- [] Implement the RFC (cc @rust-lang/XXX -- can anyone write up mentoring
      instructions?)
- [ ] Adjust documentation ([see instructions on rustc-dev-guide][doc-guide])
- [ ] Stabilization PR ([see instructions on rustc-dev-guide][stabilization-guide])

[stabilization-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#stabilization-pr
[doc-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#documentation-prs

### Unresolved Questions

- Maybe feature gate should be `generic_const_array_size`?

XXX --- list all the "unresolved questions" found in the RFC to ensure they are
not forgotten

### Implementation history

Implemented in #106281

**Comment by lcnr** [comment]
I don't think we should add many hacks to the current transmute check and don't believe that an implementation which does so should be stabilized. I would instead expect to also handle this as part of the `BikeshedIntrinsicFrom` trait.

**Comment by dvdhrm** [comment]
FYI: The feature gate is called `transmute_generic_consts` (rather than wrongly referenced in title and summary: `transmute_const_generics`).

**Comment by zachs18** [comment]
I think the part of https://github.com/rust-lang/rust/pull/106281 that actually constructed `SizeSkeleton::Generic` was effectively reverted by https://github.com/rust-lang/rust/pull/122820, so I'm not sure that `SizeSkeleton::Generic` actually does anything currently? (i.e. it seems like it is never constructed)

(This is unrelated to https://github.com/rust-lang/rust/pull/114009 whose change is also gated under this feature)
